### PR TITLE
feat: implement highlight reaction features

### LINKED ISSUE #882: feat: add reactions to highlights
This was shipped to beta as a non working feature. I am also not sure if the it works in the API, but we will need to track reactions and list the reactions under the highlight. 

The user should also be forced to log in to make a reaction. 

Screen Shot 2023-02-20 at 9 16 44 AM

### PR DESCRIPTION
**Pull request by OgDev-01**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
This PR: 

- Adds highlight reaction implementation
- Adds [react-emoji-render](https://github.com/tommoor/react-emoji-render) to render emojis

## Related Tickets & Documents
Closes #882

## Mobile & Desktop Screenshots/Recordings
image

image

### Mobile
image

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by 0-vortex** [comment]
early review: emoji approach is reversed: https://github.com/github/g-emoji-element - this is the hardest part of the PR IMHO 😅

**Comment by 0-vortex** [comment]
Use https://github.com/tommoor/react-emoji-render with `:${emojiName}:` and fallback from our url rather than the https://github.com/github/g-emoji-element if you want to get ahead quickly

**Review by bdougie — CHANGES_REQUESTED** [review]
Can we force a login when adding a reaction while not logged in? 

Screen Shot 2023-04-11 at 10 57 36 AM

**Comment by 0-vortex** [comment]
```
Error: 404 
{"statusCode":404,"message":"You are not following this user","error":"Not Found"}
```

Please fix the console spam, can't test on the deploy preview :<

Screenshot 2023-04-11 at 19 57 21

**Comment by bdougie** [comment]
Thanks for adding the redirect. Can we redirect back to the highlights feed after log in too? It takes you to the "create insights" page.

File: components/molecules/ContributorHighlight/contributor-highlight-card.tsx [hunk 1]
```diff
@@ -79,6 +83,10 @@ const ContributorHighlightCard = ({
 
   const { follow, unFollow, isError } = useFollowUser(user);
 
+  const { data: emojis, getEmojiNameById } = useFetchAllEmojis();
```
Comment (brandonroberts) [inline components/molecules/ContributorHighlight/contributor-highlight-card.tsx:86]: This should be moved outside of this component to the page itself and passed in as a prop so its only fetched/shared once for each highlight card on the page.

**Review by brandonroberts — CHANGES_REQUESTED** [review]

**Comment by OgDev-01** [comment]
> Thanks for adding the redirect. Can we redirect back to the highlights feed after log in too? It takes you to the "create insights" page.

The context of redirecting user's to their previous location before logging in requires a little more work... I've opened an issue on that here https://github.com/open-sauced/insights/issues/1075

File: components/molecules/ContributorHighlight/contributor-highlight-card.tsx (hunk 1 above)
Comment (OgDev-01) [reply components/molecules/ContributorHighlight/contributor-highlight-card.tsx:86]: When you say move outside, Do you mean it for all three reactions-related hooks or just the `useFetchEmojis`?

**Comment by brandonroberts** [comment]
We need to update the design to account for multiple selected emojis

image

**Review by brandonroberts — APPROVED** [review]

**Review by brandonroberts — APPROVED** [review]
👍

**Review by bdougie — APPROVED** [review]

**Comment by OgDev-01** [comment]
> Mobile view needs to wrap emojis
> 
> image

Thanks for fishing that out

**Comment by 0-vortex** [comment]
quick solution for shipping any layout in 50% of the time without errors:
1. flex
2. mobile-first

**Pull request merged**